### ISSUE TITLE
solve returning nan intead of the solution

### ISSUE THREAD

**Issue by Polyacene** [symengine/symengine#1745]
Dear All,
in writing an example we stumbled over the following issue:
```
#include <iostream>

#include <symengine/expression.h>
#include <symengine/solve.h>

// Maxima:
// (%i1) eq1 : t + (-1/2)*B*U - B*e = 0;
//                                          B U
// (%o1)                          t - B e - --- = 0
//                                           2
// (%i2) eq2 : (1/2)*A*U - A*e + B*U = 0;
//                                             A U
// (%o2)                       (- A e) + B U + --- = 0
//                                              2
// (%i3) solve([eq1,eq2],[A,B]);
//                                 4 U t          2 t
// (%o3)                   [[A = ---------, B = -------]]
//                                  2    2      2 e + U
//                               4 e  - U


/// g++ bug.C -o bug -l symengine -l mpfr -l gmp -lflint && ./bug  

int main(const int argc, const char *argv[])
{
	int order = (argc > 1) ? atoi(argv[1]) : 5;
	
	auto t = SymEngine::Expression("t");
	auto A = SymEngine::symbol("A");
	auto B = SymEngine::symbol("B");
	auto U = SymEngine::Expression("U");
	auto e = SymEngine::Expression("ek");
	
	///------------------------------
	/// equations
	///------------------------------
	
	SymEngine::Expression eq1 = t - B * U / 2 - B * e;
	SymEngine::Expression eq2 =     A * U / 2 - A * e + B * U;
    
	///------------------------------
	/// solve
	///------------------------------
    
    /// o.k.: 
    auto sols = SymEngine::linsolve({eq1, eq2}, {B, A});
    std::cout << "\nB & A: "  << std::endl;
    for( auto& s : sols)
        std::cout << *s << std::endl;

    /// false:
    sols = SymEngine::linsolve({eq1, eq2}, {A, B});
	std::cout << "\nA & B: "  << std::endl;
    for( auto& s : sols)
        std::cout << *s << std::endl;
}
```
leading to
```
B & A: 
-t/((-1/2)*U - ek)
U*t/(((1/2)*U - ek)*((-1/2)*U - ek))

A & B: 
nan
-t/((-1/2)*U - ek)
```

Presumably it is related to eq1 being independent from `A`, but that's just guessing.
I would also like to suggest to add an example like this to the examples page, as it takes a while to find the syntax for solving equations. Nevertheless is really nice and readable.

**Comment by isuruf** [comment]
Thanks for the report. You are right that it's because `eq1` is independent from `A`. There was no pivoting done when solving the matrix. Can you send your example as a test case in https://github.com/symengine/symengine/blob/9fc2716cab4a6d2d89a3f9d765a04ef1594c6bcf/symengine/tests/basic/test_solve.cpp#L386 as a PR? I'll try and fix symengine then.

**Comment by Polyacene** [comment]
I  guess I should not use the nice expression syntax?

**Comment by Polyacene** [comment]
I've now changed the example to
```
    auto x = SymEngine::symbol("x"), y = SymEngine::symbol("y");
    auto a = SymEngine::symbol("a"), c = SymEngine::symbol("c");
    
    auto eqns = { add({mul(a, y), c}), add({mul(a, x), mul(a, y)})};
    auto s1 = div(c, a);
    auto s2 = neg(s1);
    
    std::cout << "s1: " << *s1 << " s2: " << *s2 << std::endl;
    for( auto& s : eqns )
        std::cout << "\neqns: " << *s << std::endl;
    
    auto solns1 = linsolve( eqns, {y, x});
    for( auto& s : solns1 )
        std::cout << "\nsolns1: " << *s << std::endl;

    auto solns2 = linsolve( eqns, {x, y});
    for( auto& s : solns2 )
        std::cout << "\nsolns2: " << *s << std::endl;
```

I have to say using expressions is much nicer.

**Comment by HQS-Peter** [comment]
I switched to my company account and tried to add a new branch, issue-1745, in order to submit a pull request.
Sadly, I was not able to push it:
```
remote: Permission to symengine/symengine.git denied to HQS-Peter.
```
I'm using git on a daily basis on out company's git server, however I've never submitted a PR on github.
Is the idea to create a new branch correct and I have to look into access issues, or is a different workflow assumed?

**Comment by lkeegan** [comment]
I think the usual github workflow would be to first fork the symengine repository (there should be a "fork" button in to the top right of the page), then you can check out your fork, add a branch, and submit this as a PR.

**Comment by HQS-Peter** [comment]
o.k., Seemingly I'm not allowed to perform a fork.
So I have to wait until I'm allowed to do. 
I have added a unit test as suggested and it behaves as expected.

**Comment by HQS-Peter** [comment]
Strange, I had the message `You cannot fork because you own this repository and are not a member of any organizations`.
Someonelse from HQS forked it, and now it also works for me. I have no clue what's going on.

I'm now compiling and checking everything and try to submit a PR.

**Comment by HQS-Peter** [comment]
@isuruf  I've now submitted a PR.

**Comment by isuruf** [comment]
> I guess I should not use the nice expression syntax?

You should if you'd like to. Rest of the tests predate `Expression` class.

**Comment by HQS-Peter** [comment]
o.k., too late, I already had submitted the PR. Next time.
Anything else that I should do?

**Comment by lkeegan** [comment]
resolved by #1746
